plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -125,15 +125,16 @@ func isVolumeConflict(volume api.Volume, pod *api.Pod) bool {
 // - AWS EBS forbids any two pods mounting the same volume ID
 // - Ceph RBD forbids if any two pods share at least same monitor, and match pool and image.
 // TODO: migrate this into some per-volume specific code?
-func NoDiskConflict(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+func NoDiskConflict(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	predicateFails := make([]algorithm.PredicateFailureReason, 0)
 	for _, v := range pod.Spec.Volumes {
 		for _, ev := range nodeInfo.Pods() {
 			if isVolumeConflict(v, ev) {
-				return false, ErrDiskConflict
+				return false, append(predicateFails, ErrDiskConflict), nil
 			}
 		}
 	}
-	return true, nil
+	return len(predicateFails) == 0, predicateFails, nil
```

Comment:
Or actually, you should
return true, nil, nil
here.

We should avoid allocations whenever it's possible.